x/tokenomics/keeper/keeper_test.go

```diff
@@ -0,0 +1,136 @@
+	cmd.InitSDKConfig()
+}
+
+type TestSuite struct {
+	suite.Suite
+
+	sdkCtx  sdk.Context
+	ctx     context.Context
+	keepers keepertest.TokenomicsModuleKeepers
+	claim   prooftypes.Claim
+}
+
+func (s *TestSuite) SetupTest() {
+	supplierAddr := sample.AccAddress()
+	appAddr := sample.AccAddress()
+
+	// Prepare and insert the claim
+	s.claim = prooftypes.Claim{
+		SupplierAddress: supplierAddr,
+		SessionHeader: &sessiontypes.SessionHeader{
+			ApplicationAddress:      appAddr,
+			Service:                 &sharedtypes.Service{Id: testServiceId},
+			SessionId:               "session_id",
+			SessionStartBlockHeight: 1,
+			SessionEndBlockHeight:   sessionkeeper.GetSessionEndBlockHeight(1),
+		},
+		RootHash: []byte("default_roo_hash"),
+	}
+
+	s.keepers, s.ctx = keepertest.NewTokenomicsModuleKeepers(s.T())
+	s.sdkCtx = sdk.UnwrapSDKContext(s.ctx)
+}
+
+func TestSettleExpiringSuite(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+func (s *TestSuite) TestClaimSettlesWhenAProofIsRequired() {
+	// Retrieve default values
+	t := s.T()
+	ctx := s.ctx
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	claim := s.claim
+
+	// Add the claim & verify it exists
+	s.keepers.UpsertClaim(ctx, claim)
+	claims := s.keepers.GetAllClaims(ctx)
+	s.Require().Len(claims, 1)
+
+	// Settle expiring claims at height 2 (while the session is still active).
+	// Expectations: No claims should be settled because the session is still ongoing
+	sdkCtx = sdkCtx.WithBlockHeight(claim.SessionHeader.SessionEndBlockHeight - 2)
+	numClaimsSettled, numClaimsExpired, err := s.keepers.SettlePendingClaims(sdkCtx)
+	// Check that no claims were settled
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), numClaimsSettled)
+	require.Equal(t, uint64(0), numClaimsExpired)
+	// Validate that the claim still exists
+	claims = s.keepers.GetAllClaims(ctx)
+	require.Len(t, claims, 1)
+
+	// Try to settle expiring claim a little after it ended.
+	// Goal: Claims should not be settled because the proof window hasn't closed yet.
+	sdkCtx = sdkCtx.WithBlockHeight(claim.SessionHeader.SessionEndBlockHeight + 2)
+	numClaimsSettled, numClaimsExpired, err = s.keepers.SettlePendingClaims(sdkCtx)
+	// Check that no claims were settled
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), numClaimsSettled)
+	require.Equal(t, uint64(0), numClaimsExpired)
+	// Validate that the claim still exists
+	claims = s.keepers.GetAllClaims(ctx)
+	require.Len(t, claims, 1)
+
+	// Try to settle expiring claims a long time after it ended
+	// Expectation: All (1) claims should be settled.
+	sdkCtx = sdkCtx.WithBlockHeight(claim.SessionHeader.SessionEndBlockHeight * 10)
+	numClaimsSettled, numClaimsExpired, err = s.keepers.SettlePendingClaims(sdkCtx)
+	// Check that no claims were settled
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), numClaimsSettled)
+	require.Equal(t, uint64(0), numClaimsExpired)
+	// Validate that the claims expired
+	claims = s.keepers.GetAllClaims(ctx)
+	require.Len(t, claims, 0)
```

Comment:
I'm confused about what seem like inconsistencies to me in this section. The first comment indicates that "all claims should be settled" but the second says "no claims were settled"; is this just copypasta? If so, are the assertions that follow the second comment also copypasta? 

Additionally, the third comment and final assertion seems to indicate that the claim is expected to expire but I would've expected that to have an impact on the `numClaimsExpired` assertion. Am I missing somehting?

Finally, this test is called `TestClaimSettlesWhenAProofIsRequired` but base on the above, it doesn't seem like it's actually exercising a scenario which includes a settled proof.